### DISCUSSION TITLE
What are recommendations to be used instead of startMaximized configuration?

### DISCUSSION THREAD

**Discussion by dulciscrustulam** [selenide/selenide#2611] [Q&A] [answered]
Hello all!

Have a question regarding **Configuration.startMaximized**, as it was removed in 6.0.1 version, curious, how do you handle and make your life easier when debug some WEB tests?
Know that it is better to use **browserSize**  & **headless = true** but it is fine when your test is ready, but honestly saying often while debugging you really miss this **startMaximized** option that was helping you in debug with headed/headful mode. 
Also, browserSize does not work in headed mode, 
So any thoughts? Maybe developers of Selenide may suggest anything?

**Comment by asolntsev** [comment]
Hi.
When you are developing or debugging tests, you can just add this line in the beginning of tests:

```java
Configuration.browserSize = "4096x2048";
```

What means `browserSize does not work in headed mode`? Sure, it works in all modes.

**Reply by dulciscrustulam** [reply]
I use: Selenide 7.0.2
Browser: Chrome - latest ver
OS: Windows 10
Framework: JUnit5

If I have the following setup in my code:

```java
@BeforeEach
public void setup() {
  Configuration.headless=true;
  Configuration.browserSize="2560x1440";
  open(url);
}
``` 

When in test I print to the logs the info about my size resolution of the window via:
```java
log.info("Browser size rez: " + getWebDriver().manage().window().getSize());
```

I will get:
> Browser size rez: (2560, 1440)

so all is fine when headless is `true`.

But if I either have:
```java
@BeforeEach
public void setup() {
  /*Configuration.headless=true;*/
  Configuration.browserSize="2560x1440";
  open(url);
}
```

OR without, as when we assume default = false value:

```java
@BeforeEach
public void setup() {
  Configuration.browserSize="2560x1440";
  open(url);
}
``` 

OR

```java
@BeforeEach
public void setup() {
  Configuration.headless=false;
  Configuration.browserSize="2560x1440";
  open(url);
}
``` 

the output is:
> Browser size rez: (1555, 763)

And in general, I just see via my eyes, when I run headed mode that browser size setting was not applied.

Tried `1920x1080` and other sizes -> in headles=true all is good, in headed - does not work.

In case I am making somewhere any mistake, please correct me.
Thanks!

**Reply by asolntsev** [reply]
@mmarfushaa Ok, now I understand. 
I just tried on my machine, and I got similar result:
> Browser size rez:  (1512, 944)  // in headful mode
> Browser size rez:  (2560, 1440) // in headless mode

BUT
the **browser was opened at the full screen size**. Isn't it what you need?

The reason is that browser window cannot be larger than your screen. 
The resulting `(1512, 944)` is the size of my screen. The browser just cannot be bigger than that.

**Reply by dulciscrustulam** [reply]
Hello! Thank you for your answer.

I got your point but for me if I:
headed mode + Configuration.browserSize="XXXXxYYYY"
Where XXXXxYYYY is:

1280x720
1920x1080
4096x2048 etc

at the result I will get always the browser launched with the view (schema, unfortunately I can not attach any work related screenshots) like this:
![image](https://github.com/selenide/selenide/assets/36933215/1faf7c51-d557-4fcf-9487-7a4b9f49e0ab)

**where a part (in most cases footer) will be 'outside' the screen window**

I played today with even: 640x480 and 1366x768 (smaller results than I was getting in my log (1555, 763)), and when I ran:

- headless=false + set browser size to the 1367x763 ------> 1367x763 (good!)
- headless=false + set browser size 640x480 -------> got 1367x763  (re-ran several times, always 1367x763)

and always, the window will be the same as I explained via a schema above.
E.g. for 640x480 I was expecting smth like this: 
![image](https://github.com/selenide/selenide/assets/36933215/110d3d51-5df1-441c-8331-12d7260d8247)

But nope, I got 1367x763 somehow and the browser window positioned as on the first picture.

Anyway, why my question appeared there. I do not think I am doing smth wrong in my setup and in case for you it works at least - fine. But in case I am not only one with such situation, Configuration.startMaximized was a very nice approach that was reducing this 'fun' time of trying to find any suitable  Configuration.browserSize setup while debugging as it was setting clearly the window based on my screen size.

So in case any future features may be planned and my question will be remembered - still hope this setting will be back. Or maybe you will be able to recreate my case to confirm it may happen.

**Reply by rosolko** [reply, marked as answer]
Hi, you can just place this code inside some method (for example before all block) after open method call, or even right near selenide configuration, it will initialize browser instance without any url opened.

`WebDriverRunner.getAndCheckWebDriver().manage().window().maximize();`

**Reply by dulciscrustulam** [reply]
It works for sure, but it's Selenium, so means,'back to father Selenium' :) 
Anyway, an as old config option is not planned to be back for Selenide how I got from the answers, then fine, thanks for your thoughts and time anyway.
